jwst/lib/set_telescope_pointing.py

```diff
@@ -1516,8 +1518,12 @@ def populate_model_from_siaf(model, siaf):
 
     # For TSO exposures, also populate XREF_SCI/YREF_SCI keywords,
     # which are used by the Cal pipeline to determine the
-    # location of the source
-    if model.meta.visit.tsovisit:
+    # location of the source.
+    # Note that we use a combination of the is_tso function and
+    # a check on EXP_TYPE, because there are rare corner cases
+    # where EXP_TIME=NRC_TSGRISM, TSOVISIT=False, NINTS=1, which
+    # normally return False, but we want to treat it as TSO anyway.
+    if is_tso(model) or model.meta.exposure.type.lower() in TSO_EXP_TYPES:
```

Comment:
Since `NRS_BRIGHTOBJ` is also in `TSO_EXP_TYPES` and it doesn't need those two keywords perhaps change this line to
`if is_tso(model) or model.meta.exposure.type.lower() in ['NRC_TSIMAGE', NRS_TSGRISM']"`
It may be confusing to have those two keywords in NRS_BRIGHTOBJ files.

Reply:
Instead of picking and choosing which of the various TSO modes to put XREF_SCI and YREF_SCI into, I thought it would be simpler and more consistent to just put them in *all* TSO modes. We need them for all TSO imaging and we also need them for at least NRC_TSGRISM on the spectroscopic side, so why not just put them into all TSO exposures.

Reply:
IMHO they are a kludge and in the case of `NRS_BRIGHTOBJ` only create clutter. Their meaning is (at least for NRC_TSGRISM) that the source is located there which isn't true for `NRS_BRIGHTOBJ` and may be confusing to end users.

Reply:
OK boss, I'll fix it in the next commit.

Reply:
done